Source/NSFileManager.m

```diff
@@ -2821,13 +2821,15 @@ - (BOOL) _copyFile: (NSString*)source
   int		wbytes;
   char		buffer[bufsize];
 
-  /* Assumes source is a file and exists! */
-  NSAssert1 ([self fileExistsAtPath: source],
-    @"source file '%@' does not exist!", source);
-
   attributes = [self fileAttributesAtPath: source traverseLink: NO];
-  NSAssert1 (attributes, @"could not get the attributes for file '%@'",
-    source);
+  if (attributes == nil)
+    {
+      return [self _proceedAccordingToHandler: handler
```

Comment:
This should probably call the delegate method `fileManager:shouldProceedAfterError:copyingItemAtURL:toURL:` if it exists, otherwise `fileManager:shouldProceedAfterError:copyingItemAtPath:toPath:
`, neither of which appear to be called by this method.  The accompanying `NSError` probably wants a code along the lines of `NSFileReadNoSuchFileError`.

Ideally, we would pass the `NSError` returned by `attributesOfItemAtPath:error:`.

Reply:
Is that in addition to or instead of the handler method? I understand that Apple have deprecated the handler methods in Cocoa but don't know what the state is in GNUstep.

Reply:
The implementation of the deprecated `-copyPath:toPath:handler:` method should call back to the handler (though it's not clear what happens if the hander returns `YES` to `fileManager:shouldProceedAfterError:` - your code doesn't appear to handle it?).  The implementation of `-copyItemAtURL:toURL:error:` should set the error correctly.  I think fixing this properly needs some refactoring, though if @rfm is happy then this solution is less bad than what was there already.